pages/index.js

```diff
@@ -1,27 +1,38 @@
 import Head from "next/head";
 import styled from "styled-components";
-import { v4 as uuid } from "uuid";
+import { useState } from "react";
+import { getAllTrips } from "../services/tripService";
 import TripList from "../components/Trip/TripList";
+import Loader from "../components/Modals/Loader";
 
-import useLocalStorage from "../hooks/useLocalStorage";
-import { dummyTrips } from "../db";
+export async function getServerSideProps() {
+  const tripsDb = await getAllTrips();
 
-export default function Home() {
-  const [trips, setTrips] = useLocalStorage("trips", dummyTrips);
+  return {
+    props: { tripsDb: tripsDb },
+  };
+}
+
+export default function Home({ tripsDb }) {
+  const [trips, setTrips] = useState(tripsDb);
+  const [loader, setLoader] = useState(false);
 
-  const onSubmitNewTrip = (event) => {
+  const toggleLoader = () => setLoader((loader) => !loader);
+
+  const onSubmitNewTrip = async (event) => {
     event.preventDefault();
     const countryName = event.target.country.value;
 
-    setTrips((trips) => {
-      return [
-        ...trips,
-        {
-          id: uuid().slice(0, 8),
-          country: countryName,
-        },
-      ];
+    toggleLoader();
+    const res = await fetch("/api/trips", {
+      method: "POST",
+      body: countryName,
     });
+    const newTrips = await res.json();
+    toggleLoader();
+
+    if (newTrips.error) return alert(newTrips.error);
+    setTrips(newTrips);
```

Comment:
You still use the state to render the trips on this page? I am not sure, if I unterstand it right. Is it always synchronous with your database?

Reply:
That's how i trigger a re-render with new data. It's better than a page reload.